MLAPI/MonoBehaviours/Core/NetworkedBehaviour.cs

```diff
@@ -921,7 +922,76 @@ private bool SetDirtyness()
             }
             return dirty;
         }
-        
+
+        #endregion
+
+        #region NetworkedVar
+
+        bool networkedVarInit = false;
+        const string networkedVarTypeName = "NetworkedVar`1";
+        List<INetworkedVar> networkedVarFields = new List<INetworkedVar>();
+        internal void NetworkedVarInit()
+        {
+            if (networkedVarInit)
+                return;
+            networkedVarInit = true;
+
+            FieldInfo[] sortedFields = GetType().GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.FlattenHierarchy | BindingFlags.Instance).OrderBy(x => x.Name).ToArray();
+            for (int i = 0; i < sortedFields.Length; i++)
+            {
+                Type fieldType = sortedFields[i].FieldType;
+                if (fieldType.Name == networkedVarTypeName)
+                {
+                    Type genericTypeDefinition = typeof(NetworkedVar<>);
+                    Type genericType = genericTypeDefinition.MakeGenericType(fieldType.GetGenericArguments());
+                    INetworkedVar instance = (INetworkedVar)Activator.CreateInstance(genericType, true);
+
+                    sortedFields[i].SetValue(this, instance);
+                    instance.SetNetworkedBehaviour(this);
+
+                    networkedVarFields.Add((INetworkedVar)instance);
+                }
+            }
+        }
+
+        internal void HandleNetworkedVarChangedByRemote(BitReader reader)
+        {
+            ushort index = reader.ReadUShort();
+            if (index >= networkedVarFields.Count)
+            {
+                if (LogHelper.CurrentLogLevel <= LogLevel.Normal) LogHelper.LogWarning("NetworkedVar index out of range");
+                return;
+            }
+
+            networkedVarFields[index].HandleValueChangedByRemote(reader);
+        }
+
+        internal void SendNetworkedVar(INetworkedVar networkedVar, BitWriter varWriter)
+        {
+            using (BitWriter writer = BitWriter.Get())
+            {
+                writer.WriteUShort(GetNetworkedVarIndex(networkedVar));
+                writer.WriteWriter(varWriter);
+
+                if (isClient)
```

Comment:
Don't use isClient, isClient returns true for hosts which are actually servers. Ijstead, base the logic from isServer.

Reply:
Makes sense, but that seems like a bit of a semantic pitfall for users of MLAPI. We were talking about the special 'magic' of reflection instantiating NetworkedVars being a bit unintuitive for developers. Not being able to rely on isCilent to see whether you're a client is much worse than that IMO. But not an easy problem to solve I understand.